shuffle/cli/src/transactions.rs

```diff
@@ -0,0 +1,227 @@
+pub async fn handle(network: Url, tailing: bool) -> Result<()> {
+    let home = Home::new(get_home_path().as_path())?;
+    let address_str = fs::read_to_string(home.get_latest_address_path())?;
+    let address = AccountAddress::from_str(address_str.as_str())?;
+    let client = reqwest::Client::new();
+
+    let path = network.join(format!("accounts/{}/transactions", address).as_str())?;
+    let seq_num = get_sequence_number(&client, &network, address).await?;
+    let start = max(seq_num as i64 - 10, 0);
+
+    let resp = client
+        .get(path.as_str())
+        .query(&[("start", start.to_string().as_str())])
+        .query(&[("limit", "10")])
+        .send()
+        .await?;
+
+    let json: serde_json::Value = serde_json::from_str(resp.text().await?.as_str())?;
+
+    let all_transactions = json
+        .as_array()
+        .ok_or_else(|| anyhow::anyhow!("Failed to get transactions"))?;
+
+    for txn_index in 0..all_transactions.len() {
+        write_into(&mut io::stdout(), &json[txn_index])?;
+        println!();
+        println!();
+    }
+
+    let mut old_seq_num = 0;
+    if !all_transactions.is_empty() {
+        old_seq_num = all_transactions
+            .get(all_transactions.len() - 1)
+            .ok_or_else(|| anyhow!("Couldn't get last transaction"))?["sequence_number"]
+            .to_string()
+            .replace('"', "")
+            .parse::<u64>()?;
+    }
+    if tailing {
+        // listening for incoming transactions
+        loop {
+            thread::sleep(time::Duration::from_millis(1000));
+            let resp = get_account_txns_response(&client, &path, old_seq_num, 100).await?;
+            let json_with_txns: serde_json::Value =
+                serde_json::from_str(resp.text().await?.as_str())?;
+            let size_of_json = json_with_txns
+                .as_array()
+                .ok_or_else(|| anyhow!("Couldn't convert to array"))?
+                .len();
+
+            // checking if there are transactions
+            if size_of_json > 0 {
+                let last_txn_seq_num = &json_with_txns
+                    .get(size_of_json - 1)
+                    .ok_or_else(|| anyhow!("Couldn't get last transaction"))?["sequence_number"]
+                    .to_string()
+                    .replace('"', "")
+                    .parse::<u64>()?;
+                old_seq_num = write_out_txns(
+                    json_with_txns,
+                    *last_txn_seq_num,
+                    old_seq_num,
+                    &mut io::stdout(),
+                )?;
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn write_out_txns<W: Write>(
+    json: Value,
+    last_txn_seq_num: u64,
+    old_seq_num: u64,
+    mut stdout: W,
+) -> Result<u64> {
+    // Only want to write the first transaction (seq num: 0) or
+    // transactions where the sequence number changes
+    if last_txn_seq_num == 0 || last_txn_seq_num > old_seq_num {
```

Comment:
Conceptually, I don't understand why we need a special case for 0. On the idea of single responsibility principle ?(SRP), rather than having this method figure out which txns to print, do that outside or in another method and have this method simple write the slice of txns it receives.

Reply:
Will handle the last_txn_seq_num == 0  outside of this method.